packages/client-sdk-web/src/internal/data-client.ts

```diff
@@ -0,0 +1,208 @@
+    this.clientWrapper = new cache.ScsClient(
+      `https://${props.credentialProvider.getCacheEndpoint()}`,
+      null,
+      {
+        unaryInterceptors: this.interceptors,
+      }
+    );
+  }
+
+  public async get(
+    cacheName: string,
+    key: string | Uint8Array
+  ): Promise<CacheGet.Response> {
+    try {
+      validateCacheName(cacheName);
+    } catch (err) {
+      return new CacheGet.Error(normalizeSdkError(err as Error));
+    }
+    this.logger.trace(`Issuing 'get' request; key: ${key.toString()}`);
+    const result = await this.sendGet(cacheName, key);
+    this.logger.trace(`'get' request result: ${result.toString()}`);
+    return result;
+  }
+
+  private async sendGet(
+    cacheName: string,
+    key: string | Uint8Array
+  ): Promise<CacheGet.Response> {
+    const request = new _GetRequest();
+    request.setCacheKey(key);
+    const metadata = this.createMetadata(cacheName);
+
+    return await new Promise(resolve => {
+      this.clientWrapper.get(
+        request,
+        {
+          ...this.authHeaders,
+          ...metadata,
+        },
+        (err, resp) => {
+          if (resp) {
+            switch (resp.getResult()) {
+              case ECacheResult.MISS:
+                resolve(new CacheGet.Miss());
+                break;
+              case ECacheResult.HIT:
+                resolve(new CacheGet.Hit(resp.getCacheBody_asU8()));
+                break;
+              case ECacheResult.INVALID:
+              case ECacheResult.OK:
+                resolve(
+                  new CacheGet.Error(new UnknownError(resp.getMessage()))
+                );
+                break;
+              default:
+                resolve(
+                  new CacheGet.Error(
+                    new UnknownError(
+                      'An unknown error occurred: ' + resp.getMessage()
+                    )
+                  )
+                );
+                break;
+            }
+          } else {
+            resolve(new CacheGet.Error(cacheServiceErrorMapper(err)));
+          }
+        }
+      );
+    });
+  }
+
+  public async set(
+    cacheName: string,
+    key: string | Uint8Array,
+    value: string | Uint8Array,
+    ttl?: number
+  ): Promise<CacheSet.Response> {
+    try {
+      validateCacheName(cacheName);
+    } catch (err) {
+      return new CacheSet.Error(normalizeSdkError(err as Error));
+    }
+    if (ttl && ttl < 0) {
+      return new CacheSet.Error(
+        new InvalidArgumentError('ttl must be a positive integer')
+      );
+    }
+    const ttlToUse = ttl || this.defaultTtlSeconds;
+    this.logger.trace(
+      `Issuing 'set' request; key: ${key.toString()}, value length: ${
+        value.length
+      }, ttl: ${ttlToUse.toString()}`
+    );
+
+    return await this.sendSet(cacheName, key, value, ttlToUse);
+  }
+
+  private async sendSet(
+    cacheName: string,
+    key: string | Uint8Array,
```

Comment:
grpc web is nice cause the generated request objects can take in either string or Uint8Array, so no need for a `convert` function

Reply:
Nice!